controllers/edgeconfig_controller.go

```diff
@@ -0,0 +1,240 @@
+				return true
+			},
+		}).
+		WithOptions(controller.Options{MaxConcurrentReconciles: r.MaxConcurrentReconciles}).
+		Complete(r)
+}
+
+func createPlaybookExecutions(edgeConfig *v1alpha1.EdgeConfig) []v1alpha1.PlaybookExecution {
+	playbookExecutions := []v1alpha1.PlaybookExecution{}
+	for _, peEdgeConfig := range edgeConfig.Spec.EdgePlaybook.Playbooks {
+		var playbookExec v1alpha1.PlaybookExecution
+		playbookExec.ObjectMeta.Name = edgeConfig.Name
+		playbookExec.ObjectMeta.OwnerReferences = []v1.OwnerReference{{
+			APIVersion: edgeConfig.APIVersion,
+			Kind:       edgeConfig.Kind,
+			Name:       edgeConfig.Name,
+			UID:        edgeConfig.UID,
+		}}
+		playbookExec.ObjectMeta.Namespace = edgeConfig.Namespace
+		playbookExec.Spec.Playbook = peEdgeConfig
+		playbookExec.Spec.ExecutionAttempt = 0
+		playbookExecutionStatus := v1alpha1.PlaybookExecutionStatus{
+			Conditions: []v1alpha1.PlaybookExecutionCondition{
+				{
+					Type:   v1alpha1.PlaybookExecutionDeploying,
+					Status: v1.ConditionTrue,
+				},
+			},
+		}
+
+		playbookExec.Status = playbookExecutionStatus
+		playbookExecutions = append(playbookExecutions, playbookExec)
+	}
+
+	return playbookExecutions
+}
+
+func (r *EdgeConfigReconciler) addPlaybookExecutionToDevices(ctx context.Context, edgeConfig *v1alpha1.EdgeConfig, edgeDevices []v1alpha1.EdgeDevice) error {
+	playbookExecutionBases := createPlaybookExecutions(edgeConfig)
+	f := func(ctx context.Context, devices []v1alpha1.EdgeDevice) []error {
+
+		var errs []error
+		// access the item in the iterable directly instead of using the iterator variable
+		// to fix implicit memory aliasing in for loop
+		for i := range devices {
+			select {
+			case <-ctx.Done():
+				errs = append(errs, fmt.Errorf("context canceled: %w", ctx.Err()))
+				return errs
+			default:
+			}
+			edgeDevice := devices[i]
+			if !r.hasPlaybookExecution(edgeDevice, edgeDevice.Name+"-"+edgeConfig.Name) {
+				patch := client.MergeFrom(edgeDevice.DeepCopy())
+				for _, playbookExecutionBases := range playbookExecutionBases {
+					playbookExecution := playbookExecutionBases.DeepCopy()
+					playbookExecution.Name = edgeDevice.Name + "-" + edgeConfig.Name
+
+					peStatus :=
+						v1alpha1.PlaybookExec{Name: playbookExecution.Name,
+							PlaybookExecutionStatus: playbookExecution.Status}
+					edgeDevice.Status.PlaybookExecutions = append(edgeDevice.Status.PlaybookExecutions, peStatus)
+
+					err := r.PlaybookExecutionRepository.Create(ctx, playbookExecution)
+					if err != nil && errors.IsAlreadyExists(err) {
+						errs = append(errs, err)
+						continue
+					}
+
+					if err != nil {
+						errs = append(errs, err)
+						continue
+					}
+
+					err2 := r.EdgeDeviceRepository.PatchStatus(ctx, &edgeDevice, &patch)
+					if err2 != nil {
+						errs = append(errs, err2)
+						continue
+					}
+				}
+			}
+		}
+		return errs
+	}
+
+	errs := r.executeConcurrent(ctx, f, edgeDevices)
+	if len(errs) != 0 {
```

Comment:
I would use here go multiple-error as we used on device-worker :-)

Reply:
If you don't have a strong opinion I'd prefer to use array of errors for the moment and create a separate PR (if needed) to address this suggestion.

This code uses methods already defined and used by other controllers. That change will impact several files, so I'd like to do not increase the complexity of this PR (that already contains a lot of changes).